bot/glossary_command.go

```diff
@@ -0,0 +1,149 @@
+package bot
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/quintilesims/slackbot/db"
+	"github.com/quintilesims/slackbot/models"
+	glob "github.com/ryanuber/go-glob"
+	"github.com/urfave/cli"
+)
+
+// NewGlossaryCommand returns a cli.Command that manages !glossary
+func NewGlossaryCommand(store db.Store, w io.Writer) cli.Command {
+	return cli.Command{
+		Name:  "!glossary",
+		Usage: "manage the glossary",
+		Subcommands: []cli.Command{
+			{
+				Name:      "define",
+				Usage:     "add or set an entry in the glossary",
+				ArgsUsage: "KEY DEFINITION",
+				Action:    newGlossaryDefineAction(store, w),
+			},
+			{
+				Name:      "rm",
+				Usage:     "remove an entry from the glossary",
+				ArgsUsage: "KEY",
+				Action:    newGlossaryRemoveAction(store, w),
+			},
+			{
+				Name:      "search",
+				Usage:     "search for entries in the glossary",
+				ArgsUsage: "GLOB",
+				Action:    newGlossarySearchAction(store, w),
+			},
+		},
+	}
+}
+
+func newGlossaryDefineAction(store db.Store, w io.Writer) func(c *cli.Context) error {
+	return func(c *cli.Context) error {
+		args := c.Args()
```

Comment:
[nit] I think [this](https://github.com/quintilesims/slackbot/blob/master/bot/interview_command.go#L172) is a little bit cleaner way to grab/check the arguments; It would also be nice to use a consistent pattern.

Reply:
Cool, I was wondering about this.

Reply:
Should I just do `c.Args().Tail()` for definition then"